### DISCUSSION TITLE
Sign in with different provider

### DISCUSSION THREAD

**Discussion by sas604** [nextauthjs/next-auth#3769] [Help] [answered]
I have a basic auth with multiple providers (google/github/gitlab) users are persistent on MongoDB. The problem I have is that if a user signed in with, say, Gitlab provider later attempts to sign in with any other provider will cause an OAuthAccountNotLinked error. Is this behavior expected, and are there any ways to link users to multiple accounts?

**Comment by baptisteArno** [comment, marked as answer]
Hey @sas604 , [check this out](https://next-auth.js.org/faq#security): "When I sign in with another account with the same email address, why are accounts not linked automatically?"

**Reply by sas604** [reply]
Thank you, I didn’t think about it from this perspective.
